addons/account/models/account_move.py

```diff
@@ -3130,17 +3156,55 @@ def _compute_has_reconciled_entries(self):
     def _autopost_draft_entries(self):
         ''' This method is called from a cron job.
         It is used to post entries such as those created by the module
-        account_asset.
+        account_asset and recurring entries created by this very method.
         '''
         records = self.search([
             ('state', '=', 'draft'),
             ('date', '<=', fields.Date.context_today(self)),
-            ('auto_post', '=', True),
-        ])
-        for ids in self._cr.split_for_in_conditions(records.ids, size=1000):
-            self.browse(ids)._post()
-            if not self.env.registry.in_test_mode():
-                self._cr.commit()
+            ('auto_post', '!=', 'no'),
+        ], limit=100)
+        records._post()
+        records.filtered(lambda r: r.auto_post != 'at_date')._copy_recurring_entries()
+
+        if len(records) == 100:  # assumes there are more whenever search hits limit
+            self.env.ref('account.ir_cron_auto_post_draft_entry')._trigger()
+
+    def _copy_recurring_entries(self):
```

Comment:
actually, only called by the cron but nothing prevents the user to create and post manually a recurring invoice. It will then be ignored totally by the system :thinking:

Reply:
Absolutely. See my comments below about your suggestion of creating all entries at once.

Reply:
Now fixed by moving the recurring logic to the _post() method